test/test_autograd.py

```diff
@@ -3219,6 +3219,50 @@ def backward(ctx, x):
         # in the same thread recursively
         DeepReentrant.apply(v).sum().backward()
 
+    def test_reentrant_priority(self):
+        order = []
+
+        class MyFunction(Function):
+            @staticmethod
+            def forward(ctx, x):
+                return x
+
+            @staticmethod
+            def backward(ctx, x):
+                order.append("MyFunction")
+                return x
+
+        class Reentrant(Function):
+            @staticmethod
+            def forward(ctx, x):
+                with torch.enable_grad():
+                    ctx.x = Variable(x.data, requires_grad=True)
+                    ctx.x = ctx.x - 1
+                return ctx.x.detach()
```

Comment:
This is a pretty funky forward function. (It's funky because we don't normally do autograd operations inside of a forward function.) It seems like it's doing two things: you want to `return x` (forward is just identity), but you also want to create a leaf variable on context with some non-trivial autograd history. Is there a reason `x` has to be used in both cases? I'll keep reading and see if I can figure out why you create leaf variables in forward ;)

Reply:
Oh I see, you're also using `ctx.x` to keep track about how many times you recurse.